src/wallet/wallet.js

```diff
@@ -262,10 +293,36 @@ export class Wallet {
   encryptRecoveryData (kdfParams, recoveryKeypair) {
     let recoveryWallet = new Wallet(
       this.email,
-      this._keypair,
+      this._signingKeypair,
       recoveryKeypair.accountId()
     )
 
     return recoveryWallet.encrypt(kdfParams, recoveryKeypair.secret())
   }
+
+  /**
+   * Set signing Keypair
+   *
+   * @param {Keypair} keypair signing Keypair.
+   */
+  useSigningKeypair (keypair) {
+    if (this._isValidKeypair(keypair) && isString(keypair)) {
+      this._signingKeypair = Keypair.fromSecret(keypair)
+    } else {
+      this._signingKeypair = keypair
+    }
+  }
+
+  _isValidKeypair (keypair) {
```

Comment:
Oh

binary methods should behave either in returning `true/false` or throwing/not throwing an error. In second case it should be called smth like `checkIfKeypairIsValid`, cause people expect a function with name starting from `is...` to exactly return false when a condition haven't passed.